go-controller/hybrid-overlay/pkg/controller/node.go

```diff
@@ -29,7 +29,7 @@ func nodeChanged(node1 *kapi.Node, node2 *kapi.Node) bool {
 // 1) the node's hybrid overlay hostsubnet
 // 2) the node's first InternalIP
 // 3) the node's distributed router MAC (eg the MAC to which VXLAN packets should be sent)
-func getNodeDetails(node *kapi.Node) (*net.IPNet, net.IP, net.HardwareAddr) {
+func getNodeDetails(node *kapi.Node, queryDrMac bool) (*net.IPNet, net.IP, net.HardwareAddr) {
```

Comment:
`queryDRMAC`, but really should it be `requireDRMAC` and return an error if `requireDRMAC` is `true` and the annotation isn't set, but no error if `requireDRMAC` is `false` and the annotation isn't set?

Reply:
@danwinship I tried it with returning an error and its ugly. We really just want to log the invalid address and return nil so that we can more easily compare old vs. new. So I split the function into two so that we don't have the requireDRMAC thing. I think that should address your concern.